pkg/server/server.go

```diff
@@ -417,22 +417,15 @@ func writeKubeConfig(certs string, config *Config) error {
 }
 
 func setupDataDirAndChdir(config *config.Control) error {
-	var (
-		err error
-	)
-
-	config.DataDir, err = ResolveDataDir(config.DataDir)
-	if err != nil {
-		return err
-	}
+	var err error
 
 	dataDir := config.DataDir
 
-	if err := os.MkdirAll(dataDir, 0700); err != nil {
+	if err = os.MkdirAll(dataDir, 0700); err != nil {
```

Comment:
What requires this change? In general, these values should remain scoped within the `if` context as they don't live on past it.

Reply:
If anything, remove line 420 entirely. There's generally no reason to share a single error value throughout.